test/e2e/testsuites/pre_provisioned_dangling_attach_volume_tester.go

```diff
@@ -0,0 +1,98 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testsuites
+
+import (
+	"context"
+	"fmt"
+
+	"sigs.k8s.io/azuredisk-csi-driver/pkg/azuredisk"
+	"sigs.k8s.io/azuredisk-csi-driver/test/e2e/driver"
+	"sigs.k8s.io/azuredisk-csi-driver/test/utils/azure"
+	"sigs.k8s.io/azuredisk-csi-driver/test/utils/credentials"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+// PreProvisionedDanglingAttachVolumeTest will provision required PV(s), PVC(s) and Pod(s)
+// Testing that a volume could be reattached to a different node on DanglingAttachError
+type PreProvisionedDanglingAttachVolumeTest struct {
+	CSIDriver       driver.PreProvisionedVolumeTestDriver
+	AzureDiskDriver *azuredisk.Driver
+	Pod             PodDetails
+	VolumeID        string
+	VolumeContext   map[string]string
+}
+
+func (t *PreProvisionedDanglingAttachVolumeTest) Run(client clientset.Interface, namespace *v1.Namespace) {
+
+	// Setup required PV(s) and PVC(s) corresponding to PodDetails that will be shared among two pods
+	tpod, cleanup := t.Pod.SetupWithPreProvisionedVolumes(client, namespace, t.CSIDriver, t.VolumeContext)
+	// Defer must be called here for PV(s) and PVC(s) to be removed after the test completion and not earlier
+	for i := range cleanup {
+		defer cleanup[i]()
+	}
+
+	// Get diskURI, diskName and disk's resourceGroup
+	pvcname := tpod.pod.Spec.Volumes[0].VolumeSource.PersistentVolumeClaim.ClaimName
+	diskURI, err := tpod.GetVolumeDiskURI(pvcname)
+	framework.ExpectNoError(err, fmt.Sprintf("Error getting uri for azuredisk %v", err))
+	diskName, err := azuredisk.GetDiskName(diskURI)
+	framework.ExpectNoError(err, fmt.Sprintf("Error getting diskName for azuredisk %v", err))
+	resourceGroup, err := azuredisk.GetResourceGroupFromURI(diskURI)
+	framework.ExpectNoError(err, fmt.Sprintf("Error getting resourceGroup for azuredisk %v", err))
+
+	// Get azureClient
+	creds, err := credentials.CreateAzureCredentialFile()
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	azureClient, err := azure.GetAzureClient(creds.Cloud, creds.SubscriptionID, creds.AADClientID, creds.TenantID, creds.AADClientSecret)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	// Get VMSS
+	vmss, err := azureClient.ListVMSS(context.Background(), resourceGroup)
+	framework.ExpectNoError(err, fmt.Sprintf("Error getting the vmss list: %v", err))
+
+	// Attach the disk to 0 VM instance of VMSS
+	ginkgo.By("attaching disk to node#0")
+	err = azureClient.AttachDiskToVMSSVMInstance(context.Background(), resourceGroup, *vmss[0].Name, "0", diskName, t.VolumeID)
+	framework.ExpectNoError(err, fmt.Sprintf("Error attaching disk to node#0: %v", err))
+
+	// get a list of available nodes for scheduling the pod
+	nodes := tpod.ListNodes()
+
+	// we need at least 2 nodes to verify the reattach volume
+	if len(nodes) > 2 {
```

Comment:
`>= 2`, but really this should make the opposite check and fail in the body of the statement instead of raising the failure in the `else` clause.